src/tests/ftest/deployment/ior_per_rank.py

```diff
@@ -0,0 +1,134 @@
+        """
+
+        self.log.info("Running Test on rank: %s", rank)
+        # create the pool on specified rank.
+        self.add_pool(connect=False, target_list=[rank])
+
+        # execute ior on given rank for different transfer sizes and collect the results
+        for idx, _ in enumerate(self.transfer_sizes):
+            try:
+                self.ior_cmd.transfer_size.update(self.transfer_sizes[idx])
+                self.ior_cmd.flags.update(self.write_flags)
+                dfs_out = self.run_ior_with_pool(fail_on_warning=self.log.info)
+                dfs_perf_write = IorCommand.get_ior_metrics(dfs_out)
+                self.ior_cmd.flags.update(self.read_flags)
+                dfs_out = self.run_ior_with_pool(create_cont=False, fail_on_warning=self.log.info)
+                dfs_perf_read = IorCommand.get_ior_metrics(dfs_out)
+
+                # Destroy container, to be sure we use newly created container in next iteration
+                self.container.destroy()
+                self.container = None
+
+                # gather actual and expected perf data to be compared
+                if idx == 0:
+                    dfs_max_write = float(dfs_perf_write[0][IorMetrics.MAX_MIB])
+                    dfs_max_read = float(dfs_perf_read[0][IorMetrics.MAX_MIB])
+                    actual_write_x = percent_change(dfs_max_write, self.expected_bw)
+                    actual_read_x = percent_change(dfs_max_read, self.expected_bw)
+                else:
+                    dfs_max_write = float(dfs_perf_write[0][IorMetrics.MAX_OPS])
+                    dfs_max_read = float(dfs_perf_read[0][IorMetrics.MAX_OPS])
+                    actual_write_x = percent_change(dfs_max_write, self.expected_iops)
+                    actual_read_x = percent_change(dfs_max_read, self.expected_iops)
+
+                # compare actual and expected perf data
+                self.assertLessEqual(abs(actual_write_x), self.write_x,
+                                     "Max Write Diff too large for rank: {}".format(rank))
+                self.assertLessEqual(abs(actual_read_x), self.read_x,
+                                     "Max Read Diff too large for rank: {}".format(rank))
+                # collect list of good nodes
+                good_node = self.server_managers[0].get_host(rank)
+                if ((good_node not in self.good_nodes) and
+                        (good_node not in self.failed_nodes.keys())):
+                    self.good_nodes.append(good_node)
+            except (TestFail, DaosTestError) as _error:
+                # collect bad nodes
+                failed_node = self.server_managers[0].get_host(rank)
+                if failed_node not in self.failed_nodes:
+                    self.failed_nodes[failed_node] = [rank]
+                else:
+                    self.failed_nodes[failed_node].append(rank)
+                if failed_node in self.good_nodes:
+                    self.good_nodes.remove(failed_node)
```

Comment:
To avoid this complexity, you could just keep two lists / one dictionary of good/bad ranks, and then call `self.server_managers[0].get_host(rank)` when you display them, e.g.
```suggestion
                self.rank_status['passed'].append(rank)
            except (TestFail, DaosTestError) as _error:
                self.rank_status['failed'].append(rank)
```
```
        self.log.info("List of good nodes:")
        for rank in self.rank_status['passed']:
            self.log.info("  rank: %s, host: %s", rank, self.server_managers[0].get_host(rank))
```